packages/rc-app/settings/settings.ts

```diff
@@ -44,3 +44,165 @@ export const settings: ISetting[] = [
         packageValue: "",
     },
 ];
+
+export const propSettings: ISetting[] = [
```

Comment:
question: can we add validation to each setting?

Reply:
Validation in the sense @sidmohanty11  ?

Reply:
for example, width values can only be CSS width values. Maybe we can zod validator or something similar to check for the correctness of the arbitrary string

Reply:
We can't use external npm packages in RC app development for now. Most likely, it will not compile. I have tried integrating external packages while making RC app prototypes, and it throws errors. It is in the experimental phase for now.

https://developer.rocket.chat/docs/npm-package-support-experimental

Reply:
can we validate it by creating util functions on the app side? Not adding any extra package just some basic ones.

for example for width we can check if the value is a number and contains px or vw or % etc?

Reply:
I don't think we can able validation in rc app settings, it doesn't accept any rules or anything that can check this as far as i know, but I'll surely check once and ask someone if this is possible !!